### PR TITLE
MM-24041: make aws instance types configurable

### PR DESCRIPTION
**Pull request by agnivade**
#### Summary

Make aws instance types configurable

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-24041

### CODE REVIEW

File: docs/deployer_config.md [hunk 1]
```diff
@@ -12,12 +12,30 @@ The name of the cluster. This will be prefixed to all resources in AWS that are
 
 The number of Mattermost application instances.
 
-## AgentCount
+## AppInstanceType
+
+*int*
```
Comment (isacikgoz) [inline docs/deployer_config.md:17]: In several occurrences we should use 👇 

```suggestion
*string*
```

Comment (agnivade) [reply docs/deployer_config.md:17]: copypasta errors ..

**Comment by agnivade** [comment]
Hehe .. I knew you would ask this. Yes, I saw that and intentionally kept it like that because of Terraform. Terraform mentions "instance_class" for DB, and "instance_type" for EC2. That's they way it is :man_shrugging: So I kept it consistent with terraform.

**Comment by streamer45** [comment]
> Hehe .. I knew you would ask this. Yes, I saw that and intentionally kept it like that because of Terraform. Terraform mentions "instance_class" for DB, and "instance_type" for EC2. That's they way it is  So I kept it consistent with terraform.

How does AWS call it? No need to follow other's inconsistencies imho :stuck_out_tongue:

**Comment by agnivade** [comment]
I guess aws calls them types. But it just looks odd here: `instance_class             = var.db_instance_class`. This would become `instance_class             = var.db_instance_type`.

**Comment by streamer45** [comment]
> I guess aws calls them types. But it just looks odd here: `instance_class = var.db_instance_class`. This would become `instance_class = var.db_instance_type`.

Looks great to me :D. This is for the user really. If you want to keep the terraform variable as class it's okay but I'd like the configuration one to be type.

**Comment by agnivade** [comment]
Sounds good.

**Comment by agnivade** [comment]
Wait, I thought you wanted me to change the config field ?

**Comment by streamer45** [comment]
> Wait, I thought you wanted me to change the config field ?

Yes, just approving so you can merge once that's done.

**Pull request merged**